rasterio/coords.py

```diff
@@ -38,16 +43,47 @@ def disjoint_bounds(bounds1, bounds2):
     ``True`` if bounds are disjoint,
     ``False`` if bounds overlap
     """
-    bounds1_north_up = bounds1[3] > bounds1[1]
-    bounds2_north_up = bounds2[3] > bounds2[1]
-
-    if not bounds1_north_up and bounds2_north_up:
-        # or both south-up (also True)
-        raise ValueError("Bounds must both have the same orientation")
-
-    if bounds1_north_up:
-        return (bounds1[0] > bounds2[2] or bounds2[0] > bounds1[2] or
-                bounds1[1] > bounds2[3] or bounds2[1] > bounds1[3])
-    else:
-        return (bounds1[0] > bounds2[2] or bounds2[0] > bounds1[2] or
-                bounds1[3] > bounds2[1] or bounds2[3] > bounds1[1])
+    try:
+        intersect_bounds(bounds1, bounds2)
+        return False
+    except BoundsError:
+        return True
+
+
+def reorient_bounds(bounds):
+    def order(a, b):
+        if b < a:
+            return b, a
+        return a, b
+
+    l, b, r, t = bounds
+    l, r = order(l, r)
+    b, t = order(b, t)
+    return BoundingBox(l, b, r, t)
```

Comment:
@groutr I still don't see how this really affects users. We say up front that the datasets given to merge must have the same CRS. If they aren't, the input is garbage. We could say "Hey, garbage in, garbage out" (essentially what we have been doing) or we could fail immediately when detecting that the CRS are inconsistent.

I'm :-1: on reorienting bounding boxes because it tells users that it's okay to pass in inconsistent bounding boxes. It's going to make more work for maintainers of this project in the long run.